### DISCUSSION TITLE
Is there a `qsv` tutorial?

### DISCUSSION THREAD

**Discussion by rickhg12hs** [dathere/qsv#1189] [Q&A] [answered]
Is there a `qsv` tutorial that shows very basic operations, i.e., something that shows a newcomer how to use `qsv` to do very basic operations?

For example:
* selecting certain columns from a CSV file.
* specifying the field separation character(s)/regex
* concatenating multiple CSV files
* modifying columns of a CSV file (e.g., concatenating strings, simple math involving multiple columns, format changes)
* etc.

A tutorial for more advanced operations (if they exist) may be useful, as well.

For example:
* input from irregular CSV files
* variable assignment (for later use) from input of irregular CSV file (e.g., header line has values needed to modify following columns)
* etc.

**Comment by jqnatividad** [comment, marked as answer]
Hi @rickhg12hs ,
I'm afraid the closest thing to a tutorial right now is the [Whirlwind Tour](../blob/master/docs/whirlwind_tour.md#a-whirlwind-tour).

But as it happens, we have two things in the works!

- @rzmk is currently working on a demo site to try out qsv interactively!  The site is slated to launch mid-to-late August. 
- @a5dur is also working on some Jupyter notebooks based tutorials and they're also slated to be published this month.

As for more advanced operations, the [Cookbook](https://github.com/jqnatividad/qsv/wiki/Cookbook) and [Discussions](https://github.com/jqnatividad/qsv/discussions) are the current primary sources.

I'll try to add more meat to those sections.  Already, some power users have been adding  content to those sections and more contributions (even questions) are most welcome!

I'm hopeful that as qsv attracts more users, we can get more community members to [contribute](https://github.com/jqnatividad/qsv/blob/master/CONTRIBUTING.md) more content...

**Comment by rickhg12hs** [comment]
Thank you, I look forward to future resources.
